### ISSUE TITLE
[web] [Pointer Interceptor] App crashed on Chrome when closing a dialog containing Pointer Interceptor and Voice over is turn on

### ISSUE THREAD

**Issue by BobbyStatev** [flutter/flutter#159204]
### Steps to reproduce

1. Run the app
2. Turn Voice over on
3. Click to open the dialog
4. Close the dialog from the Enable or Disabled button

### Expected results

Navigated back to initial screen

### Actual results

Crash the app on Chrome with Error code 11

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:pointer_interceptor/pointer_interceptor.dart';

Future<void> main() async =>
    runApp(const ShowDialogExampleApp());

class ShowDialogExampleApp extends StatelessWidget {
  const ShowDialogExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: DialogExample(),
    );
  }
}

class DialogExample extends StatelessWidget {
  const DialogExample({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('showDialog Sample')),
      body: Center(
        child: OutlinedButton(
          onPressed: () => _dialogBuilder(context),
          child: const Text('Open Dialog'),
        ),
      ),
    );
  }

  Future<void> _dialogBuilder(BuildContext context) {
    return showDialog<void>(
      context: context,
      builder: (BuildContext context) {
        return AlertDialog(
          title: const Text('Basic dialog title'),
          content: const Text(
            'A dialog is a type of modal window that\n'
            'appears in front of app content to\n'
            'provide critical information, or prompt\n'
            'for a decision to be made.',
          ),
          actions: <Widget>[
            PointerInterceptor(
              child: TextButton(
                style: TextButton.styleFrom(
                  textStyle: Theme.of(context).textTheme.labelLarge,
                ),
                child: const Text('Disable'),
                onPressed: () {
                  Navigator.of(context).pop();
                },
              ),
            ),
            TextButton(
              style: TextButton.styleFrom(
                textStyle: Theme.of(context).textTheme.labelLarge,
              ),
              child: const Text('Enable'),
              onPressed: () {
                Navigator.of(context).pop();
              },
            ),
          ],
        );
      },
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>
<img width="1840" alt="Screenshot 2024-11-20 at 17 12 26" src="https://github.com/user-attachments/assets/71b54cd7-5816-49d0-880a-202fc3f8126c">
<img width="1840" alt="Screenshot 2024-11-20 at 17 13 11" src="https://github.com/user-attachments/assets/fbe3ce7a-f5e9-4423-a7f0-9481af2c631f">
<img width="1728" alt="Screenshot 2024-11-20 at 17 13 48" src="https://github.com/user-attachments/assets/c483da20-eea3-492f-b00c-3b1bcf1eb9fe">

### Logs

<details open><summary>Logs</summary>

```console
Debug service listening on ws://127.0.0.1:62102/PNkeCMZovkk=/ws
Debug service listening on ws://127.0.0.1:62102/PNkeCMZovkk=/ws
Debugger: Target crashed!
Debugger: Target crashed!
Debugger: Target crashed!
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by huycozy** [comment]
Hi @BobbyStatev 
On my end when checking this, I see the browser (Chrome) is hanging for a while and then crashed. I had to force-stop Chrome. Then I got some traces from system (MacOS) as below (it's too long so I only quote the header here), do you also notice this?

<details>
<summary>Output log</summary>

```console
Command:          Google Chrome
Path:             /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
Identifier:       com.google.Chrome
Version:          131.0.6778.70 (6778.70)
Team ID:          EQHXZ8M8AV
Is First Party:   No
Beta Identifier:  4F6EA092-864F-5E0E-A085-EC333719BFC2
Architecture:     x86_64
Parent:           launchd [1]
PID:              21520
Time Since Fork:  246583s

Event:            hang
Duration:         3.19s
Steps:            32 (100ms sampling interval)

Hardware model:   MacBookPro16,2
Active cpus:      8
HW page size:     4096
VM page size:     4096

Time Since Boot:  558896s
Time Awake Since Boot: 379886s
Time Since Wake:  14794s

Fan speed:        3357 rpm -> 3477 (+120)
Total CPU Time:   6.672s (20.9G cycles, 29.1G instructions, 0.72c/i)
Advisory levels:  Battery -> 3, User -> 2, ThermalPressure -> 1, Combined -> 2
Free disk space:  52.53 GB/465.63 GB, low space threshold 3072 MB
Vnodes Available: 77.18% (203107/263168)

Preferred User Language: en-VN, zh-Hans-VN
Country Code:     VN
Keyboards:        ABC, Arabic, German, French
OS Cryptex File Extents: 2467

--------------------------------------------------
Timeline format: stacks are sorted chronologically
Use -i and -heavy to re-report with count sorting
--------------------------------------------------


Heaviest stack for the main thread of the target process:
  32  start + 1805 (dyld + 25293) [0x7ff815b882cd]
  32  main + 205 (Google Chrome + 5229) [0x10a25746d]
  32  ChromeMain + 699 (Google Chrome Framework + 15355) [0x123097bfb]
  32  ChromeMain + 34573584 (Google Chrome Framework + 34588240) [0x125190650]
  32  ChromeMain + 34571142 (Google Chrome Framework + 34585798) [0x12518fcc6]
  32  ChromeMain + 36803969 (Google Chrome Framework + 36818625) [0x1253b0ec1]
  32  ChromeMain + 36804674 (Google Chrome Framework + 36819330) [0x1253b1182]
  32  ChromeMain + 26483019 (Google Chrome Framework + 26497675) [0x1249d928b]
  32  ChromeMain + 19978363 (Google Chrome Framework + 19993019) [0x1243a51bb]
  32  ChromeMain + 40844531 (Google Chrome Framework + 40859187) [0x12578b633]
  32  ChromeMain + 119096952 (Google Chrome Framework + 119111608) [0x12a22bfb8]
  32  -[NSApplication run] + 610 (AppKit + 199769) [0x7ff819904c59]
  32  ChromeMain + 59399088 (Google Chrome Framework + 59413744) [0x12693d4f0]
  32  ChromeMain + 940362 (Google Chrome Framework + 955018) [0x12317d28a]
  32  ChromeMain + 115545695 (Google Chrome Framework + 115560351) [0x129ec8f9f]
  32  -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1290 (AppKit + 10962612) [0x7ff81a3486b4]
  32  _DPSNextEvent + 902 (AppKit + 260991) [0x7ff819913b7f]
  32  _BlockUntilNextEventMatchingListInModeWithFilter + 66 (HIToolbox + 1006146) [0x7ff821718a42]
  32  ReceiveNextEventCommon + 646 (HIToolbox + 1005732) [0x7ff8217188a4]
  32  RunCurrentEventLoopInMode + 292 (HIToolbox + 982723) [0x7ff821712ec3]
  32  CFRunLoopRunSpecific + 550 (CoreFoundation + 499854) [0x7ff815ff908e]
  32  __CFRunLoopRun + 2752 (CoreFoundation + 504190) [0x7ff815ffa17e]
  32  __CFRunLoopDoSource1 + 534 (CoreFoundation + 509232) [0x7ff815ffb530]
  32  __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 41 (CoreFoundation + 509433) [0x7ff815ffb5f9]
  32  mshMIGPerform + 179 (HIServices + 35039) [0x7ff81c8d38df]
  32  _XCopyParameterizedAttributeValue + 594 (HIServices + 343886) [0x7ff81c91ef4e]
  32  _AXXMIGCopyParameterizedAttributeValue + 473 (HIServices + 171310) [0x7ff81c8f4d2e]
  32  ??? (CoreAccessibility + 144269) [0x7ffb2a2fa38d]
  32  ??? (CoreAccessibility + 144001) [0x7ffb2a2fa281]
  32  ??? (CoreAccessibility + 139724) [0x7ffb2a2f91cc]
  32  ??? (CoreAccessibility + 148451) [0x7ffb2a2fb3e3]
  32  ??? (CoreAccessibility + 150625) [0x7ffb2a2fbc61]
  32  ??? (CoreAccessibility + 173064) [0x7ffb2a301408]
  32  ??? (CoreAccessibility + 476318) [0x7ffb2a34b49e]
  27  ??? (CoreAccessibility + 460042) [0x7ffb2a34750a]
  27  ??? (CoreAccessibility + 206921) [0x7ffb2a309849]
  24  ??? (CoreAccessibility + 462588) [0x7ffb2a347efc]
  24  ??? (CoreAccessibility + 206921) [0x7ffb2a309849]
  12  ??? (CoreAccessibility + 260149) [0x7ffb2a316835]
  8   ??? (CoreAccessibility + 217356) [0x7ffb2a30c10c]
  8   ??? (CoreAccessibility + 214123) [0x7ffb2a30b46b]
  8   ??? (CoreAccessibility + 213323) [0x7ffb2a30b14b]
  8   ??? (libCoreAccessibilityAppKitAdditions.dylib + 29581) [0x7ffb2a2c938d]
  8   ??? (libCoreAccessibilityAppKitAdditions.dylib + 30191) [0x7ffb2a2c95ef]
  8   ??? (libCoreAccessibilityAppKitAdditions.dylib + 39742) [0x7ffb2a2cbb3e]
  8   ??? (libCoreAccessibilityAppKitAdditions.dylib + 12505) [0x7ffb2a2c50d9]
  8   _NSAccessibilityEntryPointValueForAttribute + 168 (AppKit + 3070196) [0x7ff819bc18f4]
  8   NSAccessibilityPerformEntryPointObject + 16 (AppKit + 9052286) [0x7ff81a17607e]
  6   ___NSAccessibilityEntryPointValueForAttribute_block_invoke.857 + 1478 (AppKit + 9070463) [0x7ff81a17a77f]
  6   -[NSAccessibilityAttributeAccessorInfo getAttributeValue:forObject:] + 58 (AppKit + 853210) [0x7ff8199a44da]
  4   ChromeMain + 140672088 (Google Chrome Framework + 140686744) [0x12b6bf598]
  2   ChromeMain + 140672088 (Google Chrome Framework + 140686744) [0x12b6bf598]
  1   ChromeMain + 140672064 (Google Chrome Framework + 140686720) [0x12b6bf580]
  1   ChromeMain + 146905638 (Google Chrome Framework + 146920294) [0x12bcb1366]
  1   ChromeMain + 12484105 (Google Chrome Framework + 12498761) [0x123c7f749]
  1   ChromeMain + 12484427 (Google Chrome Framework + 12499083) [0x123c7f88b]
  1   ChromeMain + 20882517 (Google Chrome Framework + 20897173) [0x124481d95]
  1   ChromeMain + 18452510 (Google Chrome Framework + 18467166) [0x12423095e]
....
```
</details>

**Comment by BobbyStatev** [comment]
> Hi @BobbyStatev On my end when checking this, I see the browser (Chrome) is hanging for a while and then crashed. I had to force-stop Chrome. Then I got some traces from system (MacOS) as below (it's too long so I only quote the header here), do you also notice this?
> 
> Output log

Yes I had a similar experience

**Comment by danagbemava-nc** [comment]
Hi @BobbyStatev, I just tested the code sample above but I can't seem to reproduce the issue. I am able to close the dialog and open it several times. Could I be missing something? Are there any more steps required? Kindly see the video below 

https://github.com/user-attachments/assets/8a7ba89f-bbf5-4bcd-bf8c-880a83af8e98

**Comment by BobbyStatev** [comment]
The problem can be reproduced only if you navigate between the items using tab key.

https://github.com/user-attachments/assets/16005c1a-8016-4738-820f-761a710f355d

**Comment by danagbemava-nc** [comment]
Thanks for the update @BobbyStatev, I can now reproduce the issue. 

Reproduced on web. 

Labeling for further investigation. 

Tested the versions below

```
stable: 3.24.5, 3.27.1
beta: 3.28.0-0.1.pre
master:  3.28.0-2.0.pre.38539
```

**Comment by flutter-zl** [comment]
I am not able to reproduce the issue with the latest version(3.33.0-1.0.pre.656) using voice over and tab navigation.

Please check https://crash-0627.web.app/ and the screencast.

https://github.com/user-attachments/assets/f9c901f8-bc51-4e68-aeaf-0e8550a02767

Hi @BobbyStatev, can you try the latest version?

**Comment by flutter-zl** [comment]
Let's close this issue. If someone is still able to reproduce this on newer versions of Flutter, please feel free to re-open the issue or comment.
